### ISSUE TITLE
Reset state when promise becomes null

### ISSUE THREAD

**Issue by bodograumann** [posva/vue-promised#39]
I feel like currently the behavior regarding `:promise="null"` is inconsistent.

According to the readme, the properties `isPending` and `isDelayOver` of the`#combined` slot are “reset […] when the `promise` prop changes”. This is not the case when the promise is changed to `null`, because the first line in the watcher is `if (!promise) return`.

My suggestion is to move this guard to after the point where the pending and delay state have been reset.

Basically what I am trying to do is:
```html
<Promised :promise="promise">
  <template #combined="{ error, isPending }">
    <template v-if="isPending">
      <Button @click="promise = doRequest()">Submit</Button>
    </template>
    <template v-else>
      <Button @click="promise = null">Reset</Button>
    </template>
  </template>
</Promise>
```

But currently the reset does not work like this.

**Comment by posva** [comment]
Maybe if promise is null, we can reset all the data properties back to their original values (in the `data` attribute) because that's their value if promise is initially null

**Comment by bodograumann** [comment]
Please see #40. Not sure why automatic linking has not worked…

**Comment by posva** [comment]
For the linking you have to reference it in the content of the PR. For example, use it as the body of one of the git messages
